alf/algorithms/rl_algorithm.py

```diff
@@ -132,11 +157,60 @@ def action_distribution_spec(self):
         """Return the action distribution spec for the action distributions."""
         return self._action_distribution_spec
 
+    @property
```

Comment:
Just add an observe() function, so the algorithm can do whatever it needs with the experience.